acceptance/tests/facts/non_root_users_without_errors.rb

```diff
@@ -0,0 +1,34 @@
+test_name "C59196: running facter as a non-root user should not produce permission errors" do
+  tag 'risk:high'
+
+  # confine :except, :platform => 'aix' # bug FACT-1586
+
+  confine :except, :platform => 'windows' # this test is for unix systems
+  confine :except, :platform => 'aix' # has errors from the su command
+  confine :except, :platform => 'cisco' # has errors from the su command
+  confine :except, :platform => 'osx' # does not support managehome
+  confine :except, :platform => 'solaris' # does not work with managehome on solaris boxes
+  confine :except, :platform => 'eos-' # does not support user creation ARISTA-37
+
+  require 'facter/acceptance/user_fact_utils'
+  extend Facter::Acceptance::UserFactUtils
+
+  agents.each do |agent|
+    non_root_user = "nonroot"
+    facter_path = on(agent, "which facter").stdout.chomp
+
+    step "Agent #{agent}: create a #{non_root_user} user to run facter with" do
+      on(agent, "puppet resource user #{non_root_user} ensure=present managehome=true shell='#{user_shell(agent)}'")
```

Comment:
Why do you have to set `managehome=true` for the user resource. I would expect issuing the `facter` command as an existing user to work regardless of whether `$HOME` existed. And not having this restriction would allow you to include solaris and osx.

Example on RHEL7
```
[root@wlg20bt1x5bngc5 ~]# puppet resource user  foo ensure=present
Notice: /User[foo]/ensure: created
user { 'foo':
  ensure => 'present',
}
[root@wlg20bt1x5bngc5 ~]# ls /home/
[root@wlg20bt1x5bngc5 ~]#
[root@wlg20bt1x5bngc5 ~]# su foo -c /opt/puppetlabs/bin/facter
aio_agent_version => 1.10.1.46
...
virtual => vmware
[root@wlg20bt1x5bngc5 ~]# echo $?
0
```